arangod/Network/NetworkFeature.cpp

```diff
@@ -79,8 +103,21 @@ void NetworkFeature::prepare() {
   _pool = std::make_unique<network::ConnectionPool>(config);
   _poolPtr.store(_pool.get(), std::memory_order_release);
 }
+  
+void NetworkFeature::start() {
+  Scheduler* scheduler = SchedulerFeature::SCHEDULER;
+  if (scheduler != nullptr) {  // is nullptr in catch tests
+    std::lock_guard<std::mutex> guard(_workItemMutex);
+    auto off = std::chrono::seconds(1);
+    _workItem = scheduler->queueDelay(RequestLane::INTERNAL_LOW, off, _gcfunc);
+  }
+}
 
 void NetworkFeature::beginShutdown() {
+  {
+    std::lock_guard<std::mutex> guard(_workItemMutex);
+    _workItem.reset();
```

Comment:
This reminds me very much of the race condition I discussed with @jsteemann: https://github.com/arangodb/arangodb/pull/8801#discussion_r276665190